libraries/AP_RangeFinder/AP_RangeFinder_BLPing.cpp

```diff
@@ -118,105 +141,64 @@ void AP_RangeFinder_BLPing::send_message(uint16_t msgid, const uint8_t *payload,
     uart->write(HIGHBYTE(crc));
 }
 
-// distance returned in reading_cm, signal_ok is set to true if sensor reports a strong signal
-bool AP_RangeFinder_BLPing::get_reading(uint16_t &reading_cm)
+PingProtocol::MessageId PingProtocol::parse_byte(uint8_t b)
 {
-    if (uart == nullptr) {
-        return false;
-    }
-
-    float sum_cm = 0;
-    uint16_t count = 0;
-
-    // read any available lines from the lidar
-    int16_t nbytes = uart->available();
-    while (nbytes-- > 0) {
-        const int16_t b = uart->read();
-        if (b < 0) {
-            break;
-        }
-        if (parse_byte(b)) {
-            count++;
-            sum_cm += distance_cm;
-        }
-    }
-
-    if (count > 0) {
-        // return average distance of readings
-        reading_cm = sum_cm / count;
-
-        // request another distance
-        send_message(BLDPIN_MSGID_DISTANCE_SIMPLE, nullptr, 0);
-
-        return true;
-    }
-
-    // no readings so return false
-    return false;
-}
-
-// process one byte received on serial port
-// returns true if a distance message has been successfully parsed
-// state is stored in msg structure
-bool AP_RangeFinder_BLPing::parse_byte(uint8_t b)
-{
-    bool got_distance = false;
-
     // process byte depending upon current state
     switch (msg.state) {
 
-    case ParseState::HEADER1:
-        if (b == BLPING_FRAME_HEADER1) {
-            msg.crc_expected = BLPING_FRAME_HEADER1;
-            msg.state = ParseState::HEADER2;
+    case ParserState::HEADER1:
+        if (b == _frame_header1) {
+            msg.crc_expected = _frame_header1;
+            msg.state = ParserState::HEADER2;
+            msg.done = false;
         }
         break;
 
-    case ParseState::HEADER2:
-        if (b == BLPING_FRAME_HEADER2) {
-            msg.crc_expected += BLPING_FRAME_HEADER2;
-            msg.state = ParseState::LEN_L;
+    case ParserState::HEADER2:
+        if (b == _frame_header2) {
+            msg.crc_expected += _frame_header2;
+            msg.state = ParserState::LEN_L;
         } else {
-            msg.state = ParseState::HEADER1;
+            msg.state = ParserState::HEADER1;
         }
         break;
 
-    case ParseState::LEN_L:
+    case ParserState::LEN_L:
         msg.payload_len = b;
         msg.crc_expected += b;
-        msg.state = ParseState::LEN_H;
+        msg.state = ParserState::LEN_H;
         break;
 
-    case ParseState::LEN_H:
+    case ParserState::LEN_H:
         msg.payload_len |= ((uint16_t)b << 8);
         msg.payload_recv = 0;
         msg.crc_expected += b;
-        msg.state = ParseState::MSG_ID_L;
+        msg.state = ParserState::MSG_ID_L;
         break;
 
-    case ParseState::MSG_ID_L:
-        msg.msgid = b;
+    case ParserState::MSG_ID_L:
```

Comment:
Keeping ParseState and msgid would have been better for this patch IMO.

Reply:
The names name is ParseState.
The structure is called `msg`, using `msgid` for `id` will return in `msg.msgid` making it redundant.